internal/keepers/service.go

```diff
@@ -265,6 +274,71 @@ Outer:
 	}
 }
 
+func (s *onDemandUpkeepService) runSamplingUpkeeps() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ch := make(chan *ethtypes.Header, 1)
+	sub, err := s.headSubscriber.SubscribeNewHead(ctx, ch)
+	if err != nil {
+		return err
+	}
+
+	heads := make(chan *ethtypes.Header)
```

Comment:
Extension of previous comment on eth specific content. The header that is returned by the subscriber should be chain generic.

Reply:
Done